andela/troupon
# [Bug #128055091] Handle integrity error when merchant signs up with n…

…o logo

#### What does this PR do?

This PR provides an alternative path for merchants who do not submit a logo with their registration thus preventing the integrity error that occurred previously.

#### Description of Task to be completed?

This task consists of:
- Supplying Advertiser's logo field with a valid default image hosted on Cloudinary.
- Create an alternative path in the view for merchants who do not submit a logo.
- Write tests for the additions above. 

While working on this task, I discovered that a 500 error occurs whenever nexmo send message fails which wasn't user-friendly. I modified the code to redirect users to the "verify merchant page"  with a message that states there was a problem sending the message. This page provides a button to resend the OTP to user's phone.

#### How should this be manually tested?

- Visit the Troupon site.
- Go to your profile page. You must have an account and be logged in to perform this step.
- On your profile page, click on the Merchant link located on the navigation pane on the left.
- Click on "Become a merchant".
- Fill the register form but leave the logo field blank.
- Submit the form.
- Confirm that you are redirected to "verify merchant page".

#### What are the relevant pivotal tracker stories?

#128055091

#### Screenshots:

![screen shot 2016-08-18 at 4 03 15 pm](https://cloud.githubusercontent.com/assets/7263503/17778837/591cd4f6-655d-11e6-8c66-ece1263be266.png)

![screen shot 2016-08-18 at 4 03 20 pm](https://cloud.githubusercontent.com/assets/7263503/17778839/59880e4c-655d-11e6-8750-c13fff34caff.png)
